include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_histogram.h

```diff
@@ -0,0 +1,547 @@
+                             const ::std::size_t __self_lidx = __self_item.get_local_id(0);
+                             const ::std::uint32_t __wgroup_idx = __self_item.get_group(0);
+                             const ::std::size_t __seg_start = __work_group_size * __wgroup_idx * __iters_per_work_item;
+                             void* __boost_mem = static_cast<void*>(&(__local_histogram[0]) + __num_bins);
+                             __func.init_SLM_memory(__boost_mem, __self_item);
+
+                             __clear_wglocal_histograms(__local_histogram, 0, __num_bins, __self_item);
+
+                             if (__seg_start + __work_group_size * __iters_per_work_item < __n)
+                             {
+                                 _ONEDPL_PRAGMA_UNROLL
+                                 for (::std::uint8_t __idx = 0; __idx < __iters_per_work_item; __idx++)
+                                 {
+                                     ::std::size_t __val_idx = __seg_start + __idx * __work_group_size + __self_lidx;
+                                     __accum_local_atomics_iter<_histogram_index_type, _atomic_address_space>(
+                                         __input, __val_idx, __local_histogram, 0, __func, __boost_mem);
+                                 }
+                             }
+                             else
+                             {
+                                 _ONEDPL_PRAGMA_UNROLL
+                                 for (::std::uint8_t __idx = 0; __idx < __iters_per_work_item; __idx++)
+                                 {
+                                     ::std::size_t __val_idx = __seg_start + __idx * __work_group_size + __self_lidx;
+                                     if (__val_idx < __n)
+                                     {
+                                         __accum_local_atomics_iter<_histogram_index_type, _atomic_address_space>(
+                                             __input, __val_idx, __local_histogram, 0, __func, __boost_mem);
+                                     }
+                                 }
+                             }
+                             __dpl_sycl::__group_barrier(__self_item);
+
+                             __reduce_out_histograms<_bin_type>(__local_histogram, 0, __bins, __num_bins, __self_item);
+                         });
+    });
+}
+
+template <::std::uint16_t __min_iters_per_work_item, typename _ExecutionPolicy, typename _Range1, typename _Range2,
+          typename _Size, typename _IdxHashFunc, typename... _Range3>
+inline auto
+__histogram_general_private_global_atomics(_ExecutionPolicy&& __exec, const sycl::event& __init_e,
+                                           ::std::uint16_t __work_group_size, _Range1&& __input, _Range2&& __bins,
+                                           const _Size& __num_bins, _IdxHashFunc __func, _Range3&&... __opt_range)
```

Comment:
It looks that we have "redundant" signature here, in meaning `_Range2&& __bins` already has information about number  of bins or Did I miss something?